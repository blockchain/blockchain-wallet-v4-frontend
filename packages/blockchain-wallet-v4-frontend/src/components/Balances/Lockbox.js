import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import * as bowser from 'bowser'

import { Icon, Link, Text } from 'blockchain-info-components'
import CoinDisplay from 'components/Display/CoinDisplay'
import FiatDisplay from 'components/Display/FiatDisplay'

const Wrapper = styled.div`
  display: flex;
  padding: 10px;
  max-width: 200px;
  margin-right: 25px;
  flex-direction: row;
  align-items: center;
  border-radius: 3px;
  transition: box-shadow 0.3s, opacity 0.3s;
  opacity: ${props => (props.isInactive ? 0.5 : 1)};
  background-color: ${props => props.theme['white-blue']};
  box-shadow: ${props =>
    props.isActive ? '0px 5px 30px 0px rgba(0,0,0,0.1)' : 'none'};
  cursor: ${props => (props.disableClick ? 'not-allowed' : 'pointer')};
  * {
    cursor: ${props => (props.disableClick ? 'not-allowed' : 'pointer')};
  }
  @media (max-width: 770px) {
    margin-right: 15px;
  }
  &:last-child {
    margin-right: 0;
  }
`
const IconBox = styled.div`
  padding: 5px;
  margin-right: 10px;
  border-radius: 3px;
  background-color: ${props => props.theme[props.coin]};
`
const Balance = styled.div`
  > div:first-child {
    margin-bottom: 3px;
  }
`
const isBrowserChrome = bowser.name === 'Chrome' || bowser.name === 'Chromium'

export const CurrencyItem = props => {
  return (
    <Wrapper
      onClick={props.onClick}
      isActive={props.isActive}
      isInactive={props.isInactive}
      disableClick={props.disableClick}
    >
      <IconBox coin={props.coin}>
        <Icon size='32px' color='white' name={props.icon} />
      </IconBox>
      {props.isSaved ? (
        <Balance>
          <FiatDisplay size='14px' weight={400} coin={props.coin.toUpperCase()}>
            {props.balance}
          </FiatDisplay>
          <CoinDisplay size='13px' weight={200} coin={props.coin.toUpperCase()}>
            {props.balance}
          </CoinDisplay>
        </Balance>
      ) : isBrowserChrome ? (
        <Link size='12px' weight={300}>
          <FormattedMessage
            id='components.balances.savecointolockbox'
            defaultMessage='Click here to add {coin} to your Lockbox'
            values={{ coin: props.coin.toUpperCase() }}
          />
        </Link>
      ) : (
        <Text size='12px' weight={300}>
          <FormattedMessage
            id='components.balances.savecointolockboxbrowser'
            defaultMessage='Use the Chrome browser to add {coin} to your Lockbox'
            values={{ coin: props.coin.toUpperCase() }}
          />
        </Text>
      )}
    </Wrapper>
  )
}
