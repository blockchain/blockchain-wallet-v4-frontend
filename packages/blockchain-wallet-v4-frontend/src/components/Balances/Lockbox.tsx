import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Icon, Link, Text } from 'blockchain-info-components'
import Bowser from 'bowser'
import styled from 'styled-components'

import CoinDisplay from 'components/Display/CoinDisplay'
import FiatDisplay from 'components/Display/FiatDisplay'
import { model } from 'data'
import { media } from 'services/styles'

const Wrapper = styled.div<{
  disableClick?: boolean
  isActive?: boolean
  isInactive?: boolean
}>`
  display: flex;
  padding: 10px;
  width: 165px;
  margin-right: 25px;
  flex-direction: row;
  align-items: center;
  border-radius: 3px;
  transition: box-shadow 0.3s, opacity 0.3s;
  opacity: ${props => (props.isInactive ? 0.5 : 1)};
  background-color: ${props => props.theme.grey000};
  box-shadow: ${props =>
    props.isActive ? 'rgba(0,0,0,.25) 0px 5px 12px 0px' : 'none'};
  cursor: ${props => (props.disableClick ? 'not-allowed' : 'pointer')};
  * {
    cursor: ${props => (props.disableClick ? 'not-allowed' : 'pointer')};
  }
  ${media.tablet`
    margin-right: 15px;
  `}
  &:last-child {
    margin-right: 0;
  }
`
const IconBox = styled.div<{ coin: string }>`
  padding: 5px;
  margin-right: 10px;
  border-radius: 3px;
  background-color: ${props => props.theme[props.coin]};
`
const Balance = styled.div`
  > div:first-child {
    margin-bottom: 3px;
  }
`
const browser = Bowser.getParser(window.navigator.userAgent)
const isBrowserSupported = browser.satisfies(
  model.components.lockbox.supportedBrowsers
)

export const CurrencyItem = props => {
  return (
    <Wrapper
      onClick={props.onClick}
      isActive={props.isActive}
      isInactive={props.isInactive}
      disableClick={props.disableClick}
    >
      <IconBox coin={props.coin}>
        <Icon size='32px' color='white' name={props.icon} />
      </IconBox>
      {props.isSaved ? (
        <Balance>
          <FiatDisplay size='14px' weight={500} coin={props.coin.toUpperCase()}>
            {props.balance}
          </FiatDisplay>
          <CoinDisplay size='13px' weight={400} coin={props.coin.toUpperCase()}>
            {props.balance}
          </CoinDisplay>
        </Balance>
      ) : isBrowserSupported ? (
        <Link size='11px' weight={500}>
          <FormattedMessage
            id='components.balances.savecointolockbox'
            defaultMessage='Click here to add {coin} to your Lockbox'
            values={{ coin: props.coin.toUpperCase() }}
          />
        </Link>
      ) : (
        <Text size='12px' weight={400}>
          <FormattedMessage
            id='components.balances.browserblock'
            defaultMessage='Use the Brave, Chrome, Firefox or Opera browsers to add {coin} to your Lockbox'
            values={{ coin: props.coin.toUpperCase() }}
          />
        </Text>
      )}
    </Wrapper>
  )
}
