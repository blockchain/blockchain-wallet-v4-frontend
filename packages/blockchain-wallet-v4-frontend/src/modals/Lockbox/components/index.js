import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Icon, Text } from 'blockchain-info-components'

const StepHeader = styled.div`
  background-color: ${(props) => props.theme.white};
  justify-content: space-evenly;
  padding: 30px 30px 0px 30px;
  border-radius: 4px 4px 0 0;
  flex-direction: row;
  align-items: center;
  display: flex;
`
const Step = styled.div`
  flex-direction: column;
  align-items: center;
  text-align: center;
  display: flex;
`
const Line = styled.div`
  width: 100%;
  position: relative;
  border-top: 4px solid ${(props) => props.theme.grey000};
  &:before {
    content: '';
    left: 0;
    width: 0%;
    height: 4px;
    bottom: 0px;
    position: absolute;
    background: ${(props) => props.theme.blue600};
    transition: width 0.5s 0.3s;
  }
  &.complete {
    &:before {
      width: 100%;
    }
  }
`
const Circle = styled.div`
  width: 22px;
  height: 22px;
  z-index: 10;
  display: flex;
  border-radius: 15px;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  overflow: hidden;
  border: 2px solid ${(props) => props.theme.grey000};
  transition: border 0.3s, background-color 0.2s;
  &.active,
  &.complete {
    border: 2px solid ${(props) => props.theme.blue600};
  }
  &.complete {
    background-color: ${(props) => props.theme.blue600};
  }
`
const CircleContent = styled.div`
  top: 10px;
  position: relative;
  transition: top 0.5s;
  &.complete {
    top: -12px;
  }
`
const CheckIcon = styled(Icon)`
  margin-top: 10px;
`

class ModalStepper extends React.PureComponent {
  render() {
    // currentStep & totalSteps do not follow zero-based numbering
    const { currentStep, totalSteps } = this.props

    return (
      <StepHeader>
        {[...Array(totalSteps)].map((r, i) => {
          i += 1
          const active = currentStep === i ? 'active' : null
          const complete = currentStep > i ? 'complete' : null
          return (
            <React.Fragment key={currentStep}>
              <Step>
                <Circle className={`${active} ${complete}`}>
                  <CircleContent className={`${active} ${complete}`}>
                    <Text size='13px' color={active ? 'blue600' : complete ? 'white' : 'grey000'}>
                      {i}
                    </Text>
                    <CheckIcon name='checkmark' size='10px' weight='600' color='white' />
                  </CircleContent>
                </Circle>
              </Step>
              {i !== totalSteps && <Line className={complete} />}
            </React.Fragment>
          )
        })}
      </StepHeader>
    )
  }
}

ModalStepper.propTypes = {
  currentStep: PropTypes.number.isRequired,
  totalSteps: PropTypes.number.isRequired
}

export default ModalStepper
