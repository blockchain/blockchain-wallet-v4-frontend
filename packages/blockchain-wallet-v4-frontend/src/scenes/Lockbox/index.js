import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import styled from 'styled-components'
import { withRouter, Route, Switch } from 'react-router-dom'
import Joyride, { ACTIONS, EVENTS, STATUS } from 'react-joyride'

import { actions, selectors } from 'data'
import { TourTooltip, TOUR_STEPS } from './model'
import LockboxDashboard from './Dashboard'
import LockboxOnboard from './Onboard'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 100%;
`
class LockboxContainer extends React.PureComponent {
  state = { run: false, steps: TOUR_STEPS }

  componentDidMount () {
    this.checkForcedRouting()
  }

  componentDidUpdate (prevProps) {
    if (this.props !== prevProps) {
      if (this.props.showProductTour) {
        this.onStartTour()
      }
      this.checkForcedRouting()
    }
  }

  // need to force route match on all entries from side menu click
  checkForcedRouting = () => {
    if (this.props.location.pathname === '/lockbox') {
      this.props.lockboxActions.determineLockboxRoute()
    }
  }

  onStartTour = () => {
    this.setState({ run: true })
  }

  handleTourCallbacks = data => {
    if ([EVENTS.STEP_AFTER, EVENTS.TARGET_NOT_FOUND].includes(data.type)) {
      // advance current step
      this.setState({
        stepIndex: data.index + (data.action === ACTIONS.PREV ? -1 : 1)
      })
    } else if ([STATUS.FINISHED, STATUS.SKIPPED].includes(data.status)) {
      // end tour
      this.setState({ run: false })
      this.props.lockboxActions.setProductTourVisibility(false)
    }
  }

  render () {
    const { steps, run } = this.state

    return (
      <Wrapper>
        <Joyride
          run={run}
          steps={steps}
          disableScrollParentFix={true}
          callback={this.handleTourCallbacks}
          tooltipComponent={TourTooltip}
          {...this.props.Joyride}
        />
        <Switch>
          <Route
            path='/lockbox/dashboard/:deviceIndex'
            component={LockboxDashboard}
            exact
          />
          <Route
            path='/lockbox/settings/:deviceIndex'
            component={LockboxDashboard}
            exact
          />
          <Route path='/lockbox/onboard' component={LockboxOnboard} exact />
        </Switch>
      </Wrapper>
    )
  }
}

const mapStateToProps = state => ({
  showProductTour: selectors.components.lockbox.getProductTourVisibility(state)
})

const mapDispatchToProps = dispatch => ({
  lockboxActions: bindActionCreators(actions.components.lockbox, dispatch)
})

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(LockboxContainer)
)
