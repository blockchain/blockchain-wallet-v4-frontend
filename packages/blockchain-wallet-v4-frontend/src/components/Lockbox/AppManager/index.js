import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import styled from 'styled-components'
import { FormattedHTMLMessage } from 'react-intl'

import {
  BlockchainLoader,
  Button,
  Image,
  Text
} from 'blockchain-info-components'
import { FAIL_STATUS_TIMEOUT, SUCCESS_STATUS_TIMEOUT } from './model'
import { actions, selectors } from 'data'
import * as Lockbox from 'services/LockboxService'
import LockboxAppManager from './template'
import { Remote } from 'blockchain-wallet-v4'

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
`
const ConnectStep = styled.div`
  text-align: center;
  & > :last-child {
    margin: 10px 0;
  }
`
const Loader = styled(BlockchainLoader)`
  margin: 20px;
`
const ContinueButton = styled(Button)`
  margin-top: 12px;
`
const ConnectInstructions = styled(Text)`
  text-align: center;
  padding: 5px;
  margin: 30px 0 36px;
`
const AllowManagerText = styled(Text)`
  margin: 20px 0;
  text-align: center;
`
const BtcRequiredText = styled(Text)`
  margin-top: 8px;
  text-align: center;
`

const getKeyByValue = value => {
  return Object.keys(Lockbox.constants.supportedApps).find(
    key => Lockbox.constants.supportedApps[key] === value
  )
}

class LockboxAppManagerContainer extends React.PureComponent {
  state = {}

  componentDidMount () {
    this.props.lockboxActions.initializeAppManager()
  }

  componentDidUpdate (prevProps) {
    if (this.props.appChangeStatus !== prevProps.appChangeStatus) {
      this.props.appChangeStatus.cata({
        Success: val => {
          // install/uninstall APIs use different keys for appName
          const AppName = Lockbox.constants.supportedApps[val.appName]
            ? Lockbox.constants.supportedApps[val.appName]
            : val.appName
          this.props.lockboxActions.resetAppChangeStatus()
          this.setState({ [AppName]: { changeType: '', status: 'Success' } })
          // clears the status text after 5 seconds
          setTimeout(() => {
            this.setState({ [AppName]: { status: null } })
          }, SUCCESS_STATUS_TIMEOUT)
        },
        Failure: val => {
          // install/uninstall APIs use different keys for appName
          const AppName = Lockbox.constants.supportedApps[val.appName]
            ? Lockbox.constants.supportedApps[val.appName]
            : val.appName
          this.props.lockboxActions.resetAppChangeStatus()
          this.setState({
            [AppName]: { status: 'Error', error: val.error }
          })
          // clears the status text after 10 seconds
          setTimeout(() => {
            this.setState({ [AppName]: { changeType: '', status: null } })
          }, FAIL_STATUS_TIMEOUT)
        },
        Loading: () => {},
        NotAsked: () => {}
      })
    }
  }

  componentWillUnmount () {
    this.props.lockboxActions.resetAppChangeStatus()
  }

  onAppInstall = (appName, coin) => {
    this.setState({
      [appName]: {
        status: 'Updating',
        changeType: 'Installing',
        appName
      }
    })
    this.props.lockboxActions.installApplication(coin)
  }

  onAppUninstall = appName => {
    this.setState({
      [appName]: {
        status: 'Updating',
        changeType: 'Uninstalling',
        appName
      }
    })
    this.props.lockboxActions.uninstallApplication(appName)
  }

  render () {
    const { appChangeStatus, appVersionInfos, connection } = this.props
    const disableButtons = !Remote.NotAsked.is(appChangeStatus)
    const appListView = appVersionInfos.cata({
      Success: apps => {
        const appList = apps.map(app => {
          const appName = app.name
          const coin = getKeyByValue(appName)
          return (
            <LockboxAppManager
              key={appName}
              app={app}
              coin={coin}
              installApp={() => {
                this.onAppInstall(appName, coin)
              }}
              uninstallApp={() => {
                this.onAppUninstall(appName)
              }}
              coinState={this.state[appName]}
              disableUpdates={disableButtons}
              requireBtc={this.props.newDevice}
            />
          )
        })
        return (
          <React.Fragment>
            {appList}
            <AllowManagerText size='11px' weight={300}>
              <FormattedHTMLMessage
                id='components.lockbox.appmanager.prompt'
                defaultMessage='If prompted, be sure to allow the "Device Manager" onto the device during app updates.'
              />
            </AllowManagerText>
            {this.props.newDevice && (
              <BtcRequiredText size='10px' weight={300}>
                <FormattedHTMLMessage
                  id='components.lockbox.appmanager.btcrequired'
                  defaultMessage='The Bitcoin app is needed to connect your Lockbox to your wallet.'
                />
              </BtcRequiredText>
            )}
            <ContinueButton
              disabled={disableButtons}
              onClick={this.props.onClose}
              nature='primary'
              fullwidth
            >
              {this.props.mainButtonText}
            </ContinueButton>
          </React.Fragment>
        )
      },
      Failure: () => (
        <Text size='16px' weight={300}>
          <FormattedHTMLMessage
            id='components.lockbox.appmanager.appfailure'
            defaultMessage='Failed to load application list. Please try again later.'
          />
        </Text>
      ),
      Loading: () => (
        <Wrapper>
          <Loader style={{ margin: '32px' }} width='75px' height='75px' />
        </Wrapper>
      ),
      NotAsked: () => {}
    })

    return (
      <Wrapper>
        {connection.app !== 'DASHBOARD' ? (
          <ConnectStep>
            <Image width='100%' name='lockbox-onboard-connect' />
            <ConnectInstructions size='14px' weight={300}>
              <FormattedHTMLMessage
                id='components.lockbox.appmanager.connectdevice'
                defaultMessage='Connect, unlock and open the Dashboard on your Lockbox device now.'
              />
            </ConnectInstructions>
            <ContinueButton disabled nature='primary' fullwidth>
              <FormattedHTMLMessage
                id='components.lockbox.appmanager.waiting'
                defaultMessage='Waiting...'
              />
            </ContinueButton>
          </ConnectStep>
        ) : (
          appListView
        )}
      </Wrapper>
    )
  }
}

LockboxAppManagerContainer.propTypes = {
  mainButtonText: PropTypes.element.isRequired,
  newDevice: PropTypes.bool,
  onClose: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  appChangeStatus: selectors.components.lockbox.getAppChangeStatus(state),
  appVersionInfos: selectors.components.lockbox.getLatestApplicationVersions(
    state
  ),
  connection: selectors.components.lockbox.getCurrentConnection(state)
})

const mapDispatchToProps = dispatch => ({
  lockboxActions: bindActionCreators(actions.components.lockbox, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LockboxAppManagerContainer)
