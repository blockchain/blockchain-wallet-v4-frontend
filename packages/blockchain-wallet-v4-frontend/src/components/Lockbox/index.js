import AppManager from './AppManager'

export { AppManager }
