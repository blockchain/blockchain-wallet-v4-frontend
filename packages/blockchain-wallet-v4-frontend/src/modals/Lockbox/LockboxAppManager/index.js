import React from 'react'
import PropTypes from 'prop-types'
import { compose, bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'

import { Modal, ModalBody, ModalHeader } from 'blockchain-info-components'
import { actions } from 'data'
import modalEnhancer from 'providers/ModalEnhancer'
import { AppManager } from 'components/Lockbox'

class LockboxAppManagerModal extends React.PureComponent {
  onClose = () => {
    this.props.lockboxActions.lockboxModalClose()
    this.props.closeAll()
  }

  render () {
    const { deviceIndex, position, total } = this.props
    return (
      <Modal size='small' position={position} total={total}>
        <ModalHeader onClose={this.onClose}>
          <FormattedMessage
            id='modals.lockbox.appmanager.title'
            defaultMessage='App Manager'
          />
        </ModalHeader>
        <ModalBody style={{ padding: '18px' }}>
          <AppManager
            deviceIndex={deviceIndex}
            mainButtonText={
              <FormattedMessage
                id='modals.lockbox.appmanager.close'
                defaultMessage='Close App Manager'
              />
            }
            onClose={this.onClose}
          />
        </ModalBody>
      </Modal>
    )
  }
}

LockboxAppManagerModal.propTypes = {
  deviceIndex: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  closeAll: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => ({
  lockboxActions: bindActionCreators(actions.components.lockbox, dispatch)
})

const enhance = compose(
  modalEnhancer('LockboxAppManager'),
  connect(
    null,
    mapDispatchToProps
  )
)

export default enhance(LockboxAppManagerModal)
