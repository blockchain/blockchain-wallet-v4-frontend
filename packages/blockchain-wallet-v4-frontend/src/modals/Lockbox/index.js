import LockboxAppManager from './LockboxAppManager'
import LockboxFirmware from './LockboxFirmware'
import LockboxSetup from './LockboxSetup'
import LockboxShowXPubs from './LockboxShowXPubs'
import LockboxConnectionPrompt from './LockboxConnectionPrompt'
import LockboxAuthenticityCheck from './LockboxAuthenticityCheck'

export {
  LockboxAppManager,
  LockboxAuthenticityCheck,
  LockboxConnectionPrompt,
  LockboxFirmware,
  LockboxSetup,
  LockboxShowXPubs
}
