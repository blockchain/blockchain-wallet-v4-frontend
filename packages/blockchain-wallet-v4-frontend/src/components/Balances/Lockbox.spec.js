import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { CurrencyItem } from './Lockbox'

jest.mock('blockchain-info-components', () => ({
  Banner: '',
  Text: '',
  Icon: ''
}))
describe('Lockbox Balances Component', () => {
  it('should render and match default snapshot', () => {
    const component = shallow(<CurrencyItem isSaved={true} coin='btc' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should prompt user to use Chrome if coin is not saved', () => {
    jest.mock('bowser', () => ({
      name: 'NotChrome'
    }))
    const component = shallow(<CurrencyItem isSaved={false} coin='btc' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should prompt user to user to add coin if not saved and browser Chrome', () => {
    jest.mock('bowser', () => ({
      name: 'Chrome'
    }))
    const component = shallow(<CurrencyItem isSaved={true} coin='btc' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
