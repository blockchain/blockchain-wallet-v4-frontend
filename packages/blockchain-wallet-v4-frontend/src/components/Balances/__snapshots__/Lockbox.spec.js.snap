// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lockbox Balances Component should prompt user to use Chrome if coin is not saved 1`] = `
<Lockbox__Wrapper>
  <Lockbox__IconBox
    coin="btc"
  >
    <Component
      color="white"
      size="32px"
    />
  </Lockbox__IconBox>
  <Component
    size="12px"
    weight={300}
  >
    <FormattedMessage
      defaultMessage="Use the Chrome browser to add {coin} to your Lockbox"
      id="components.balances.savecointolockboxbrowser"
      values={
        Object {
          "coin": "BTC",
        }
      }
    />
  </Component>
</Lockbox__Wrapper>
`;

exports[`Lockbox Balances Component should prompt user to user to add coin if not saved and browser Chrome 1`] = `
<Lockbox__Wrapper>
  <Lockbox__IconBox
    coin="btc"
  >
    <Component
      color="white"
      size="32px"
    />
  </Lockbox__IconBox>
  <Lockbox__Balance>
    <Connect(FiatDisplayContainer)
      coin="BTC"
      size="14px"
      weight={400}
    />
    <Connect(CoinDisplayContainer)
      coin="BTC"
      size="13px"
      weight={200}
    />
  </Lockbox__Balance>
</Lockbox__Wrapper>
`;

exports[`Lockbox Balances Component should render and match default snapshot 1`] = `
<Lockbox__Wrapper>
  <Lockbox__IconBox
    coin="btc"
  >
    <Component
      color="white"
      size="32px"
    />
  </Lockbox__IconBox>
  <Lockbox__Balance>
    <Connect(FiatDisplayContainer)
      coin="BTC"
      size="14px"
      weight={400}
    />
    <Connect(CoinDisplayContainer)
      coin="BTC"
      size="13px"
      weight={200}
    />
  </Lockbox__Balance>
</Lockbox__Wrapper>
`;
